### PR TITLE
compatibility note

### PR DESCRIPTION
**Pull request by TheRealMarnes**
Because of #1340

### CODE REVIEW

File: docs/src/adoc/index.adoc [hunk 1]
```diff
@@ -4046,6 +4046,10 @@ Then configure the Jdbi factory bean in your Spring container, e.g.:
     `@Inject` annotations on the target class instead of configuring it in
     your `beans.xml`.
 
+[NOTE]
+This module has been *experimentally* shown to be compatible
```
Comment (stevenschlansker) [inline docs/src/adoc/index.adoc]: What additional validation would we need to do to drop the '*experimentally*' here?  Spring 5 is the GA version, if we can claim to support it, we should.

Comment (TheRealMarnes) [reply docs/src/adoc/index.adoc]: don't know. It seems to me that if its basic usage works and everything compiles, it probably just works, but I'm not the expert on it so I'm not gonna state it as fact without a thorough test. I guess I would set up a jdbi-spring5 that _only_ contains tests and doublecheck tx handling and stuff there?

Comment (TheRealMarnes) [reply docs/src/adoc/index.adoc]: On that note btw, what's up with #1252?

Comment (stevenschlansker) [reply docs/src/adoc/index.adoc]: I'm not sure exactly.  Nobody in the core project understands Spring transaction management (or wants to use it) and it's a lot of code to merge that we don't really understand.  And it's not currently quite ready to ship anyway.  So it's stuck in limbo...

Comment (TheRealMarnes) [reply docs/src/adoc/index.adoc]: Hmm, yeah, I see what you mean. The first half isn't so bad, but then you get to the real spring tx integration part...

**Comment by TheRealMarnes** [comment]
Just pushed a stub of an idea I had, that currently doesn't work yet. Any idea if it's possible to just take the tests of another module and run them again in the current module as if they were a dependency? I thought the `<classifier>tests</classifier>` stuff would do just that...

File: spring5/pom.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        <version>3.6.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>jdbi3-spring5</artifactId>
+    <name>jdbi3 Spring 5</name>
+    <description>jdbi Spring 4/5 forward compatibiility testing</description>
+    <url>http://jdbi.org/</url>
+
+    <properties>
+        <dep.spring.version>5.1.3.RELEASE</dep.spring.version>
```
Comment (qualidafial) [inline spring5/pom.xml]: Does this actually succeed in changing the dependency version?

Comment (TheRealMarnes) [reply spring5/pom.xml]: according to intellij > maven > show effective pom, it does

File: spring5/src/test/java/org/jdbi/v3/spring5/TestVersion.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import org.junit.Test;
+import org.springframework.core.SpringVersion;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestVersion {
+    @Test
+    public void thisIsSpring5() {
+        assertThat(SpringVersion.getVersion())
+            .isEqualTo("5.1.3.RELEASE");
```
Comment (TheRealMarnes) [inline spring5/src/test/java/org/jdbi/v3/spring5/TestVersion.java]: @qualidafial

**Comment by TheRealMarnes** [comment]
Ok, done. Turned out to be surprisingly easy. Tests extended so they're re-run, and overrode the spring version with a test to make sure.

ez pz

**Pull request merged**